cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -127,6 +127,7 @@ using internal::ToAwsString;
 using internal::ToURLEncodedAwsString;
 
 static const char kSep = '/';
+constexpr char kDefaultEndpointEnvVar[] = "AWS_ENDPOINT";
```

Comment:
How about using `AWS_ENDPOINT_URL` instead of `AWS_ENDPOINT` because aws-sdk-cpp will support `AWS_ENDPOINT_URL` https://github.com/aws/aws-sdk-cpp/issues/2587 ?
(How about accepting `AWS_ENDPOINT_URL` in arrow-rs too?)

Reply:
Could you use the same name for constant name?
For example, `kAwsEndpointEnvVar` for this case.